server/src/main/java/org/cloudfoundry/identity/uaa/login/LoginInfoEndpoint.java

```diff
@@ -624,7 +624,8 @@ private Map.Entry<String, AbstractIdentityProviderDefinition> evaluateLoginHint(
 
     @RequestMapping(value = {"/delete_saved_account"})
     public String deleteSavedAccount(HttpServletRequest request, HttpServletResponse response, String userId) {
-        Cookie cookie = new Cookie("Saved-Account-%s".formatted(userId), "");
+        String sanitizedUserId = URLEncoder.encode(userId, UTF_8);
+        Cookie cookie = new Cookie("Saved-Account-%s".formatted(sanitizedUserId), "");
```

Comment:
We also have this [utility method](https://github.com/cloudfoundry/uaa/blob/0c0f9c2c61adfbc43453e0384b39d2e0411ebb0c/server/src/main/java/org/cloudfoundry/identity/uaa/login/AccountSavingAuthenticationSuccessHandler.java#L94-L102) called when the cookie is first set

Maybe break this out into a utility class so that we use the same encoding everywhere?